bridge/client/app/_components/ktb-project-settings/ktb-project-settings.component.ts

```diff
@@ -248,11 +252,53 @@ export class KtbProjectSettingsComponent implements OnInit, OnDestroy {
 
   public reset(): void {
     this.gitSettingsSection?.reset();
-    this.unsavedDialogState = null;
+    this.projectFormTouchedSubject.next(true);
+    this.isProjectFormUntouched = true;
+    this.hideNotification();
   }
 
   public saveAll(): void {
     this.setGitUpstream();
+    this.hideNotification();
+  }
+
+  public isProjectFormInvalid(): boolean {
+    return this.isCreateMode ? this.isProjectCreateFormInvalid() : this.isProjectSettingsFormInvalid();
+  }
+
+  public isProjectCreateFormInvalid(): boolean {
+    return (
+      !this.shipyardFile ||
+      this.projectNameForm.invalid ||
+      !this.gitData.gitFormValid ||
+      this.isCreatingProjectInProgress
+    );
+  }
+
+  public isProjectSettingsFormInvalid(): boolean {
+    return !this.gitData.gitFormValid || this.isGitUpstreamInProgress;
+  }
+
+  projectFormTouched(): void {
+    this.isProjectFormUntouched = this.projectFormTouchedSubject.asObservable();
+  }
+
+  canDeactivate(): boolean | Observable<boolean> {
+    return this.isProjectFormUntouched;
+  }
+
+  // @HostListener allows us to also guard against browser refresh, close, etc.
+  @HostListener('window:beforeunload', ['$event'])
+  showNotification($event?: BeforeUnloadEvent): void {
+    if (this.canDeactivate() !== true) {
```

Comment:
If it's an `Observable` it's always `false`. The content of the observable is not checked here.
In my opinion, I would not even use `Observables` here

Reply:
The `Observable` allows the user to decide to stay or leave the page.